src/sentry/api/serializers/rest_framework/rule.py

```diff
@@ -92,6 +94,20 @@ def validate_actions(self, value):
             raise serializers.ValidationError(u"Must select an action")
         return value
 
+    def validate(self, attrs):
+        # XXX(meredith): For rules that have the Slack integration as an action
+        # we need to check if the channel_id needs to be looked up via an async task.
+        # If the "pending_save" attribute is set we want to bubble that up to the
+        # project_rule(_details) endpoints by setting it on attrs
+        actions = attrs.get("actions")
+        if actions:
+            for action in actions:
+                if action.pop("pending_save", None):
```

Comment:
@MeredithAnya this is going to mutate the input `attrs`, is that what we want? Is there a reason we can't just look at the item without mutating `action`?

Reply:
I think it would be good to have a comment regarding why we remove this key from the dict. If there's no reason then using `.get()` would be more efficient as it has 0 writes compared to this one.